flutter_local_notifications/README.md

```diff
@@ -43,37 +43,208 @@ A cross platform plugin for displaying local notifications.
+
+
+
+## 📷 Screenshots
 
 | Android | iOS |
 | ------------- | ------------- |
 | <img height="480" src="https://github.com/MaikuB/flutter_local_notifications/raw/master/images/android_notification.png"> |  <img height="414" src="https://github.com/MaikuB/flutter_local_notifications/raw/master/images/ios_notification.png"> |
 
 
 
-## Acknowledgements
+## 👏 Acknowledgements
 
 * [Javier Lecuona](https://github.com/javiercbk) for submitting the PR that added the ability to have notifications shown daily
 * [Jeff Scaturro](https://github.com/JeffScaturro) for submitting the PR to fix the iOS issue around showing daily and weekly notifications and migrating the plugin to AndroidX
 * [Ian Cavanaugh](https://github.com/icavanaugh95) for helping create a sample to reproduce the problem reported in [issue #88](https://github.com/MaikuB/flutter_local_notifications/issues/88)
 * [Zhang Jing](https://github.com/byrdkm17) for adding 'ticker' support for Android notifications
 * ...and everyone else for their contributions. They are greatly appreciated
 
-## Getting Started
 
-The GitHub repository has an example app that should demonstrate of all the supported features of the plugin. Please check the example for more detailed code samples. If you only copy and paste the Dart code then this will not work as there's setup required for each platform. Pub also generates API docs for the latest version [here](https://pub.dartlang.org/documentation/flutter_local_notifications/latest/). Besides referring to the example app and getting started section, please ensure that you have performed the steps in the integration guide for each platform further below.
 
-The following samples will demonstrate the more commonly used functionalities.
+## 📈 Testing
+
+As the plugin class is not static, it is possible to mock and verify its behaviour when writing tests as part of your application. 
+Check the source code for a sample test suite that has been kindly implemented (_test/flutter_local_notifications_test.dart_) that demonstrates how this can be done. 
+
+If you decide to use the plugin class directly as part of your tests, the methods will be mostly no-op and methods that return data will return default values. 
+
+Part of this is because the plugin detects if you're running on a supported plugin to determine which platform implementation of the plugin should be used. If it's neither Android or iOS, then it defaults to the aforementioned behaviour to reduce friction when writing tests. If this not desired then consider using mocks. 
+
+[Named constructor](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/FlutterLocalNotificationsPlugin/FlutterLocalNotificationsPlugin.private.html) can be used to pass the platform for the plugin to resolve the desired platform-specific implementation.
```

Comment:
> Named contructors can be used to pass the platform for the plugin to resolve the desired platform-specific implementation.
This sentence is structured just fine

Adding "Note there is also" only adds more noise.
If there is a specific named constructor or whether users can create multiple named contructors I'm not sure whether ot use the plural or the singular form.

I've updated this to add "Note there is also a...."

Reply:
The main issue I was trying to highlight was how the sentence starts and suggestion given was one of many, it could've also been something like "A named constructor is available..."

Reply:
Okay I get it now, I just overlooked what named constructors are... I have just reverted it.